### PR TITLE
[ADD] connector_easypost_tracker: Create mapping for TrackingDetail and TrackingLocation

### PR DESCRIPTION
**Pull request by tedsalmon**
This pull request covers LABS-159 & LABS-160 as they are co-dependent (An event *must* have a location*).

[ADD] connector_easypost_tracker: Create mapping for TrackingDetail and TrackingLocation
* Create mapping importers / tests for Events
* Create mapping importers / tests for Locations
* Create Webhook controller
* Add Binders
* Update Global and tracker README
* Add Screenshots from Picking

[IMP] connector_easypost: Update test object
* Update `ObjDict` to object type with dict attributes. This allows me to use `getattr` without having uncatchable `KeyError` thrown at me.
* Fix `import_synchronizer` string lint
* Fix `import_synchronizer` `easy_import_data` by adding data object to correct object attribute
* Fix `import_synchronizer` Environment helper `_get_env_return_importer` not sending the UID over to `ConnectorSession`

### CODE REVIEW

File: connector_easypost_tracker/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+
+class EasypostWebhookController(http.Controller):
+
+    EVENTS = {'tracker.updated': 'easypost.stock.picking.tracking.group'}
+
+    def _get_backend(self, env):
+        return env['easypost.backend'].sudo().search([
+            ('is_default', '=', True)
```
Comment (lasley) [inline connector_easypost_tracker/controllers/main.py]: Trailing comma

File: connector_easypost_tracker/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        ])
+
+    @http.route([
+        '/api/easypost/webhook',
```
Comment (lasley) [inline connector_easypost_tracker/controllers/main.py]: As much as I prefer this URL, you don't own this namespace. You own routes under `/connector_easypost_tracker/*`

File: connector_easypost_tracker/models/stock_picking_tracking_event.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    odoo_id = fields.Many2one(
+        comodel_name='stock.picking.tracking.event',
+        string='Stock Picking Tracking Event',
+        required=True,
+        ondelete='cascade',
+    )
+
+    _sql_constraints = [
+        ('odoo_uniq', 'unique(backend_id, odoo_id)',
```
Comment (lasley) [inline connector_easypost_tracker/models/stock_picking_tracking_event.py:45]: Isn't this enforced in `easypost.binding`?

File: connector_easypost_tracker/models/stock_picking_tracking_event.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+@easypost
+class StockPickingTrackingEventImportMapper(EasypostImportMapper):
+    _model_name = 'easypost.stock.picking.tracking.event'
+
+    direct = [
+        (eval_false('message'), 'message'),
+        (eval_false('status'), 'state'),
+        (eval_false('source'), 'source'),
+        (eval_false('datetime'), 'date_created')
```
Comment (lasley) [inline connector_easypost_tracker/models/stock_picking_tracking_event.py]: trail comma

File: connector_easypost_tracker/models/stock_picking_tracking_group.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 import logging
-from openerp import models, fields
+from openerp import fields, models
```
Comment (lasley) [inline connector_easypost_tracker/models/stock_picking_tracking_group.py:6]: It's funny how much I hated the menial task of alphabetizing imports, but now that I'm used to it I can't live without them alphabetized. It's one of those can't unsee things, right!? 😆

File: connector_easypost_tracker/models/stock_picking_tracking_location.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    _easypost_model = 'Tracker'
+
+    odoo_id = fields.Many2one(
+        comodel_name='stock.picking.tracking.location',
+        string='Stock Picking Tracking Location',
+        required=True,
+        ondelete='cascade',
+    )
+
+    _sql_constraints = [
```
Comment (lasley) [inline connector_easypost_tracker/models/stock_picking_tracking_location.py:42]: I think the binding constrains this

File: connector_easypost/unit/import_synchronizer.py [hunk 7]
```diff
@@ -314,21 +315,37 @@ def run(self, odoo_binding_id):
                        self.backend_record.id)
 
 
-def _get_env_return_importer(env, model_name, backend_id):
+class ObjectDict(object):
```
Comment (lasley) [inline connector_easypost/unit/import_synchronizer.py]: Hmmmm I think maybe this needs its own file in `unit`. I feel like it's going to be lost here & it's not only related to the import (tests also use)

**Review by lasley — CHANGES_REQUESTED** [review]

File: connector_easypost_tracker/controllers/main.py (hunk 2 above)
Comment (tedsalmon) [reply connector_easypost_tracker/controllers/main.py]: 😞 I gettttt it but that's lame!

File: connector_easypost_tracker/models/stock_picking_tracking_event.py (hunk 3 above)
Comment (tedsalmon) [reply connector_easypost_tracker/models/stock_picking_tracking_event.py:45]: Nope, `backend_id` and `easypost_id` are the only unique constraint imposed there.

File: connector_easypost_tracker/models/stock_picking_tracking_location.py (hunk 6 above)
Comment (tedsalmon) [reply connector_easypost_tracker/models/stock_picking_tracking_location.py:42]: Negative, ghost rider. See my previous comment on the subject :)

**Comment by tedsalmon** [comment]
My apologies @lasley, auto-pilot took over and I squashed most of my changes 😞 . To recap, I:

* Updated the route in the controller
* Added a missing trailing comma in the controller
* Added a missing trailing comma in models/stock_picking_tracking_event.py
* Moved `ObjectDict` class to its own file in unit/

**Review by lasley — APPROVED** [review]
Thanks @t3ddftw

**Pull request merged**